test/Microsoft.AspNetCore.Routing.Tests/Constraints/StringRouteConstraintTest.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing.Constraints;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Routing.Tests.Constraints
+{
+    public class StringRouteConstraintTest
+    {
+        [Fact]
+        public void StringRouteConstraintSimpleTrueTest()
+        {
+            var constraint = new StringRouteConstraint("home");
+            var values = new RouteValueDictionary(new { controller = "home" });
+
+            var match = constraint.Match(
+              httpContext: Mock.Of<HttpContext>(),
+              route: new Mock<IRouter>().Object,
+              routeKey: "controller",
+              values: values,
+              routeDirection: RouteDirection.IncomingRequest);
+
+            // Assert
+            Assert.True(match);
+        }
+
+        [Fact]
+        public void StringRouteConstraintKeyNotFoundTest()
```

Comment:
We should also cover the case where the value is in the dictionary but is `null`